# BPBR16-Bioinformatics-using-Python-for-Biomedical-Researchers

## Repository for the [BPBR16](http://gtpb.igc.gulbenkian.pt/bicourses/BPBR16/) training course

Dates: July 11th-15th 2016<br/>
<a href="http://gtpb.igc.gulbenkian.pt/bicourses/BPBR16/" >course announcement</a><br/>
<a href="http://gtpb.igc.gulbenkian.pt/bicourses/BPBR16/timetable.html" >course program</a><br/>

 <table style="width:100%">
  <tr>
    <td> <img src="./img/igclogo.png" alt="yay" height="100" width="200"></td>
    <td><img src="./img/elixir_ita_logo.png" alt="yay" height="100" width="150"></td>
    <td><img src="./img/elixir_ptg_logo.png" alt="yay" height="100" width="170"></td>
  <tr/>
  <tr>
  <img src="./img/GTPB2015logo.png" alt="yay" height="200" width="600">
  <tr/>
</table>




Python is an object-oriented programming language that is ideal for biological data analysis. The course will start from zero knowledge, and will introduce the participants to all the basic concepts of Python such as calculating, organizing data, reading and writing files, program logic and writing larger programs. All the examples and practical sessions will focus on solving biological problems. In particular the sessions will cover:

    - working with DNA and protein sequences
    - data retrieval from files and their manipulation
    - running applications, such as BLAST, locally and from a script
    - finding motifs in sequence
    - parsing common file formats (Uniprot, GenBank, PDB, BLAST) with Biopython
    - ways to find and correct program errors

The course will be highly interactive and the students will continuously put theory into practice while learning. By the end of the course, the participants will have a good understanding of Python basics and will have acquired the skills to manage any type of bioinformatics database record and to run applications from scripts. Basic Unix/Linux skills will be provided at the beginning of the course.



#### This repository is for the course materials and it is organized as follows:

Days |Lessons \& Tutorials |
------------ | -------------|
day 1 | [Intro](day1/intro.md), [Unix/Linux command line](day1/1-Unix/Unix-Theory-BPBR16.md), [Python Shell](day1/2-Pythonshell/pythonshell.md) |
day 2 |[Python Programs](day2/1-PythonPrograms/PythonPrograms.md), [Repeating Things](day2/2-RepeatingThings/RepeatingThings.md), [Structures and Modules](day2/1-PythonPrograms/PythonStructureModulesImport.md), [File Formats](day2/3-Parsing/FileFormats.md),  [Parsing I](day2/3-Parsing/Parsing-Theory-I.md), [Parsing II](day2/3-Parsing/Parsing-Theory-II.md)|
day 3 |[Functions](day3/1-Functions/functions.md),  [Best Practices](day3/2-Debugging/BestPracticesInProgramming.md), [Error Handling](day3/2-Debugging/ErrorHandling.md),  [Data Columns](day3/3-DataColumns/DataColumns.md), [Tabular Data](day3/4-TabularData/TabularData.md) |
day 4 |[Data Searching](day4/DataSearching/DataSearching.md), [NGS pipeline](day4/Pipelines/NGS_pipeline.md), [BLAST Pipeline](day4/Pipelines/Running-BLAST_sys.argv.md), [Python libraries](day4/PythonLibraries/tasks.md) |
day 5 |[Hackaton](day5/Hackathon/HACKATHON_TASKS.md) |
=======



#Instructors:
[Allegra Via](http://bioinformaticstraining.pythonanywhere.com/) ( IBBE, CNR, Bari, IT, Elixir-IT Training Coordinator)<br/>
[Vincenza Colonna](http://www.igb.cnr.it/popgenlab) (Institute of Genetics and Biophysics, CNR, Napoli, IT, Elixir-IT Deputy Training Coordinator)
